code/__HELPERS/icons.dm

```diff
@@ -1459,3 +1459,83 @@ GLOBAL_LIST_EMPTY(friendly_animal_types)
 	animate(src, pixel_x = pixel_x + shiftx, pixel_y = pixel_y + shifty, time = 0.2, loop = duration)
 	pixel_x = initialpixelx
 	pixel_y = initialpixely
+
+/**
+ * Returns a list that contains the leftmost, rightmost, bottom and top pixels in an atom's icon
+ */
+/atom/proc/get_bounding_box()
```

Comment:
Unfortunately I just don't think this is going to work. Even with only around 300-500 calls getpixel becomes one of the most overtiming things on the server. Progress bar images need to be specifically chosen and have their parameters fed in directly, or they need to be an image with the normal progress bar below and ignore the fact that part of it might take a while to start showing.